apps/ai-content-generator/src/components/app/dialog/common-generator/CommonGenerator.tsx

```diff
@@ -26,14 +26,23 @@ const CommonGenerator = () => {
   const { setProviderData, feature, localeNames } = useContext(GeneratorContext);
 
   const [parameters, dispatch] = useReducer(generatorReducer, initialParameters);
+  const [headerHeight, setHeaderHeight] = useState(0);
+  const headerRef = useRef<HTMLElement>(null);
 
   const updateProviderData = () => {
     setProviderData({
       dispatch,
     });
   };
 
-  useEffect(updateProviderData, [dispatch, setProviderData]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we may want to just add `setProviderData` to the dependency array here, this could lead to buggy code potentially

Reply:
Currently having `setProviderData` in the dependency array is causing lots of re-renders for the `CommonGenerator` component. I'm removing it and adding this comment for now and we can come back to it to refactor.